examples/nemo_run/prune_distill/nemo_prune_kd_flow.py

```diff
@@ -14,255 +14,315 @@
-
-    return most_recent
-
-
-def get_most_recent_ckpt(directory: str):
-    """
-    Find the most recent checkpoint subdirectory in a given directory.
-
-    Args:
-        directory (str): Path to the directory to search in.
-
-    Returns:
-        str: Path to the most recent subdirectory.
-    """
-    exp_dir = Path(directory) / "default"
-    assert exp_dir.exists(), f"Experiment directory {exp_dir} does not exist"
-
-    checkpoint_dir = exp_dir / "checkpoints"
-    if checkpoint_dir.exists():
-        most_recent = get_most_recent_subdir(checkpoint_dir)
-    else:
-        most_recent = get_most_recent_subdir(exp_dir)
-        checkpoint_dir = most_recent / "checkpoints"
-        assert checkpoint_dir.exists(), f"Checkpoint directory {checkpoint_dir} does not exist"
-        most_recent = get_most_recent_subdir(checkpoint_dir)
-
-    return str(most_recent)
-
-
-def export_most_recent_ckpt(directory: str, output_path: str):
-    most_recent_ckpt = get_most_recent_ckpt(directory)
-    export_ckpt(most_recent_ckpt, "hf", output_path=output_path, overwrite=True)
-
-
-def _read_chat_template(template_path: str):
-    with open(template_path) as f:
-        return f.read().strip()
-
-
-if __name__ == "__main__":
-    args = get_args()
-
-    # # # # # CONFIGURABLE PARAMETERS # # # # #
-    GPUS = 8
-    SEQUENCE_LENGTH = 8192
-    MBS = 2
-    VAL_BATCHES = 32
-    if args.mock_run:
-        GBS = 8
-        FINETUNE_STEPS = 20
-        DISTILL_STEPS = 20
-        VAL_INTERVAL = 10
-        PRUNE_SAMPLES = 3
-        assert args.data_path is None, "Argument --data_path not used in mock mode."
-    else:
-        GBS = 2048
-        FINETUNE_STEPS = 2500
-        DISTILL_STEPS = 7500
-        VAL_INTERVAL = 500
-        PRUNE_SAMPLES = 1024
-        args.data_path = args.data_path if args.data_path is not None else "lima_processed"
-    # # # # # # # # # # # # # # # # # # # # # #
-
+def main(args):
     # Common items
-    model_name = args.recipe
-    model_module = getattr(llm, model_name)
+    exp_dir = os.path.join(args.log_dir, args.experiment)
+    model_name = args.base_recipe
+
     if args.mock_run:
         data = run.Config(
             MockDataModule,
-            global_batch_size=GBS,
-            micro_batch_size=MBS,
+            global_batch_size=DISTILL_GBS,
+            micro_batch_size=DISTILL_MBS,
             seq_length=SEQUENCE_LENGTH,
         )
     else:
         tokenizer = run.Config(
             get_nmt_tokenizer,
             library="huggingface",
-            model_name=args.tokenizer_hf,
-            chat_template=_read_chat_template(args.chat_template) if args.chat_template else None,
+            model_name=args.hf_tokenizer,
         )
         data = run.Config(
-            ChatDataModule,
-            dataset_root=args.data_path,
+            PreTrainingDataModule,
+            paths=[f"{args.data_dir}/part_{i}_text_document" for i in SUBSET_IDX],
```

Comment:
how will this work if users modify the subset indices from cmdline when preprocessing with process_climbmix.py? Perhaps we just ask for the root directory and find all `{xyz}.bin` files and infer from that?